packages/flutter/lib/src/services/clipboard.dart

```diff
@@ -3,21 +3,45 @@
 // found in the LICENSE file.
 
 
+import 'dart:typed_data';
+
 import 'package:flutter/foundation.dart';
 
 import 'system_channels.dart';
 
+/// A descriptor for the type of data contained in a [ClipboardData].
+enum ClipboardDataType {
+  /// A `text/plain` data type contained in clipboard.
+  text,
+
+  /// An image encoded as a png.
+  image,
+}
+
 /// Data stored on the system clipboard.
 ///
 /// The system clipboard can contain data of various media types. This data
 /// structure currently supports only plain text data, in the [text] property.
 @immutable
 class ClipboardData {
+  //// Deprecated? Or updated to require type.
   /// Creates data for the system clipboard.
-  const ClipboardData({ this.text });
+  const ClipboardData({ this.text }) : dataType = ClipboardDataType.text, image = null;
+
+  /// Creates image data for the system clipboard.
+  const ClipboardData.image(this.image) : dataType = ClipboardDataType.image, text = null;
```

Comment:
I don't plan to fully implement this right now, adding to help design selection for multiple data types

Reply:
The support for this will be platform dependent, android doesn't support multimedia clipboard while windows/macOS/ios supports some multimedia I believe.